### PR TITLE
Optimized async "foreach" sections + removed unused code

### PR DESCRIPTION
**Pull request by polonzer**
Additional PR for https://lykkex.atlassian.net/browse/LWDEV-9224

### CODE REVIEW

File: src/Lykke.Service.Assets.Services/AssetGroupService.cs [hunk 1]
```diff
@@ -193,18 +190,21 @@ public async Task<bool> SwiftDepositEnabledAsync(string clientId, bool isIosDevi
...
+            var clientIds = (await GetClientIdsForGroupAsync(group.Name)).ToArray();
+            var clientAssetGroupTasks =
+                clientIds.Select(clientId => _clientAssetGroupLinkRepository.UpdateAsync(clientId, group));
+            tasks.AddRange(clientAssetGroupTasks);
+            var assetGroupTasks =
+                clientIds.Select(clientId => _assetGroupClientLinkRepository.UpdateAsync(clientId, group));
+            tasks.AddRange(assetGroupTasks);
+
+            tasks.Add(_assetGroupRepository.UpdateAsync(group));
+            await Task.WhenAll(tasks.ToArray());
```
Comment (dmitrysmall) [inline src/Lykke.Service.Assets.Services/AssetGroupService.cs]: Does the order of task execution matter in this place? I mean, before we had the call to
`await _assetGroupClientLinkRepository.UpdateAsync(clientId, group);`
just after the call to
`await _clientAssetGroupLinkRepository.UpdateAsync(clientId, group);`
with the same `(clientId, group)`.
And also we had
`await _assetGroupAssetLinkRepository.UpdateAsync(assetId, group);`
at the very beginning, and the
`await _assetGroupRepository.UpdateAsync(group);`
at the end. Now the order of the execution will be unknown and may differ from time to time.

File: src/Lykke.Service.Assets.Services/AssetGroupService.cs [hunk 2]
```diff
@@ -151,18 +144,22 @@ public async Task RemoveClientFromGroupAsync(string clientId, string groupName)
...
+            var clientIds = (await GetClientIdsForGroupAsync(groupName)).ToArray();
+            var clientAssetGroupTasks =
+                clientIds.Select(clientId => _clientAssetGroupLinkRepository.RemoveAsync(clientId, groupName));
+            tasks.AddRange(clientAssetGroupTasks);
+            var assetGroupTasks =
+                clientIds.Select(clientId => _assetGroupClientLinkRepository.RemoveAsync(clientId, groupName));
+            tasks.AddRange(assetGroupTasks);
+
+            tasks.Add(_assetGroupRepository.RemoveAsync(groupName));
+            await Task.WhenAll(tasks.ToArray());
```
Comment (dmitrysmall) [inline src/Lykke.Service.Assets.Services/AssetGroupService.cs]: Please, see the comment [below](https://github.com/LykkeCity/Lykke.Service.Assets/pull/53/files#r217013674).

File: src/Lykke.Service.Assets.Repositories/Erc20Repository.cs [hunk 3]
```diff
@@ -87,7 +75,7 @@ public async Task<IEnumerable<IErc20Token>> GetAllWithAssetsAsync()
 
             return (await GetByAssetIndexesAsync(indexes))
                 // Ensure, that our indexes are not corrupted. It's important, we have already faced problems with them.
-                .Where(x => x.AssetId != null);
+                .Where(x => x?.AssetId != null);
```
Comment (dmitrysmall) [inline src/Lykke.Service.Assets.Repositories/Erc20Repository.cs]: `.Where(x => !string.InNullOrWhiteSpace(x?.AssetId);`
?

File: src/Lykke.Service.Assets.Services/AssetGroupService.cs (hunk 1 above)
Comment (polonzer) [reply src/Lykke.Service.Assets.Services/AssetGroupService.cs]: The order doesn't matter here. The only problem we can have here - the lack of transactionality. But I don't want to fix it right now. My goal is to reduce requests latency.

File: src/Lykke.Service.Assets.Services/AssetGroupService.cs (hunk 2 above)
Comment (polonzer) [reply src/Lykke.Service.Assets.Services/AssetGroupService.cs]: The order doesn't matter here. The only problem we can have here - the lack of transactionality. But I don't want to fix it right now. My goal is to reduce requests latency.

File: src/Lykke.Service.Assets.Repositories/Erc20Repository.cs (hunk 3 above)
Comment (polonzer) [reply src/Lykke.Service.Assets.Repositories/Erc20Repository.cs]: I don't think such validation is appropriate here. Whitespace in AssetId is out-of-contracts problem.

**Pull request merged**